lib/std/c.zig

```diff
@@ -295,12 +295,23 @@ pub extern "c" fn kevent(
 ) c_int;
 
 pub extern "c" fn getaddrinfo(
-    noalias node: [*:0]const u8,
-    noalias service: [*:0]const u8,
-    noalias hints: *const addrinfo,
+    noalias node: ?[*:0]const u8,
+    noalias service: ?[*:0]const u8,
+    noalias hints: ?*const addrinfo,
     noalias res: **addrinfo,
 ) EAI;
 
+test "getaddrinfo" {
```

Comment:
No tests in `c.zig` please, this change needs no test at all.

Reply:
Sounds good, in this case the test was most useful to ensure that it had the correct effect (actually compile passing in a null and return the expected results). Is there a suggested way to make these sorts of changes and validate them? I could easily enough have written a sample program and just validated it out of the code base, just curious what the expectation would be.

Reply:
The [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/) standard tells this change is correct, trivial changes to the libc declaration parameters require a proof of their correctness rather than a test.